apps/desktop/src/platform/services/desktop-settings.service.ts

```diff
@@ -291,7 +291,10 @@ export class DesktopSettingsService {
    * Sets the modal mode of the application. Setting this changes the windows-size and other properties.
    * @param value `true` if the application is in modal mode, `false` if it is not.
    */
-  async setInModalMode(value: boolean) {
-    await this.inModalModeState.update(() => value);
+  async setModalMode(value: boolean, modalPosition?: [number, number]) {
+    await this.modalModeState.update(() => ({
+      isModalModeActive: value,
+      modalPosition: modalPosition,
```

Comment:
suggestion(non-blocking): simplify
```suggestion
      modalPosition,
```